azurerm/internal/services/advisor/tests/resource_arm_advisor_suppression_test.go

```diff
@@ -0,0 +1,221 @@
+	data := acceptance.BuildTestData(t, "azurerm_advisor_suppression", "test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMAdvisorSuppressionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMAdvisorSuppression_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAdvisorSuppressionExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMAdvisorSuppression_complete(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAdvisorSuppressionExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMAdvisorSuppression_update(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAdvisorSuppressionExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMAdvisorSuppression_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAdvisorSuppressionExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func testCheckAzureRMAdvisorSuppressionExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).Advisor.SuppressionsClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("advisor Suppression not found: %s", resourceName)
+		}
+
+		id, err := parse.AdvisorSuppressionID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if resp, err := client.Get(ctx, id.ResourceUri, id.RecommendationName, id.Name); err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("bad: Advisor Suppression %q does not exist", id.Name)
+			}
+			return fmt.Errorf("bad: Get on Advisor Suppression Client: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMAdvisorSuppressionDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).Advisor.SuppressionsClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_advisor_suppression" {
+			continue
+		}
+
+		id, err := parse.AdvisorSuppressionID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if resp, err := client.Get(ctx, id.ResourceUri, id.RecommendationName, id.Name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("bad: Get on Advisor Suppression Client: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMAdvisorSuppression_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_advisor_recommendations" "test" {}
```

Comment:
for these tests is there a repeatable recommendation we can use rather they relying on the first? also when the resource is deleted will the suppression stop?